src/test/java/org/folio/rest/api/IsbnNormalizationTest.java

```diff
@@ -0,0 +1,194 @@
+  public void canSearchForFirstIsbnWithAdditionalHyphens() {
+    createRequests("Interesting Times", "Uprooted");
+
+    find("itemIsbn = 0-552-16754-1", "Interesting Times");
+  }
+
+  @SneakyThrows
+  @Test
+  public void canSearchForFirstIsbnWithAdditionalHyphenAndTruncation() {
+    createRequests("Interesting Times", "Uprooted");
+
+    find("itemIsbn = 05-5*", "Interesting Times");
+  }
+
+  @SneakyThrows
+  @Test
+  public void canSearchForSecondIsbnWithMissingHyphens() {
+    createRequests("Interesting Times", "Uprooted");
+
+    find("itemIsbn = 9780552167543", "Interesting Times");
+  }
+
+  @SneakyThrows
+  @Test
+  public void canSearchForSecondIsbnWithMissingHyphensAndTrunation() {
+    createRequests("Interesting Times", "Temeraire", "Uprooted");
+
+    find("itemIsbn = 9780*", "Interesting Times", "Temeraire");  
+  }
+
+  @SneakyThrows
+  @Test
+  public void canSearchForSecondIsbnWithAlteredHyphens() {
+    createRequests("Interesting Times", "Temeraire");
+
+    find("itemIsbn = 9-7-8-055-2167-543", "Interesting Times");
+  }
+
+  @SneakyThrows
+  @Test
+  public void cannotFindIsbnWithTailString() {
+    createRequests("Interesting Times");
+
+    find("itemIsbn = 552-16754-3");
+  }
+
+  @SneakyThrows
+  @Test
+  public void cannotFindIsbnWithInnerStringAndTruncation() {
+    createRequests("Interesting Times");
+
+    find("itemIsbn = 552*");
+  }
+
+  private void find(String cql, String ... expectedTitles) {
+    JsonObject searchBody = searchForRequests(cql);
+    assertThat(searchBody.toString(), not(""));
+    matchItemTitles(searchBody, expectedTitles);
+  }
+
+  private void matchItemTitles(JsonObject jsonObject, String[] expectedTitles) {
+    if (expectedTitles.length == 0) {
+      assertThat(jsonObject.getInteger("totalRecords"), is(0));
+      return;
+    }
+    assertThat(jsonObject.getInteger("totalRecords"), is(expectedTitles.length));
+    JsonArray requests = jsonObject.getJsonArray("requests");
+    for (String title : expectedTitles) {
+      Boolean found = requests.stream().anyMatch(request -> isTitlePresent((JsonObject) request, title));
+      assertThat(found, is(true));
+    }
+  }
+
+  private Boolean isTitlePresent(JsonObject request, String titleToMatch) {
+    JsonObject item = request.getJsonObject("item");
+    String itemTitle = item.getString("title");
+    if (itemTitle.equals(titleToMatch)) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  private JsonObject searchForRequests(String cql) {
+    CompletableFuture<Response> searchCompleted = new CompletableFuture<>();
+    try {
+      String url = requestStorageUrl("").toString() + "?query=" + urlEncode(cql);
+
+      client.get(url, TENANT_ID, responseHandler(searchCompleted));
+      Response searchResponse = searchCompleted.get(5, SECONDS);
+
+      assertThat(searchResponse.getStatusCode(), is(200));
+      return searchResponse.getJson();
+    } catch (Exception e) {
+      return new JsonObject("");
+    }
```

Comment:
What is the intent for not propagating exceptions when searching for requests?

Is it to avoid needing to define all of the exceptions that could occur in every client of this code?

Reply:
The model code I was following had the request written like this-I assumed it was because an error at this stage would indicate something wrong with the testing environment, such as a problem with the database set-up, in which case, all tests would be failing with errors.

Reply:
> The model code I was following had the request written like this

Fair enough. It can be difficult to know if that code is an exemplar of a positive pattern or not.

> I assumed it was because an error at this stage would indicate something wrong with the testing environment, such as a problem with the database set-up, in which case, all tests would be failing with errors.

That could be the case. Under those circumstances, I would want the tests to fail with an error (that would hopefully reveal that situation) rather than an assertion mismatch (that I would then need to figure out the cause for) which is what I think this code would do.

Reply:
Okay, I'll rewrite this to push up errors to the top level.